allure-generator/src/main/javascript/components/tree-view-container/TreeViewContainer.js

```diff
@@ -0,0 +1,74 @@
+import './styles.scss';
+import {View} from 'backbone.marionette';
+import template from './TreeViewContainer.hbs';
+import {behavior, className, on, regions} from '../../decorators';
+import NodeSorterView from '../node-sorter/NodeSorterView';
+import NodeSearchView from '../node-search/NodeSearchView';
+import StatusToggleView from '../status-toggle/StatusToggleView';
+import TreeView from '../tree/TreeView';
+import {Model} from 'backbone';
+import {getSettingsForTreePlugin} from '../../utils/settingsFactory';
+
+@className('tree')
+@behavior('TooltipBehavior', {position: 'bottom'})
+@regions({
+    search: '.pane__search',
+    sorter: '.tree__sorter',
+    filter: '.tree__filter',
+    content: '.tree__content'
+})
+class TreeViewContainer extends View {
+    template = template;
+
+    initialize({routeState, tabName, baseUrl, settings = getSettingsForTreePlugin(baseUrl)}) {
+        this.state = new Model();
+        this.routeState = routeState;
+        this.baseUrl = baseUrl;
+        this.tabName = tabName;
+        this.listenTo(this.routeState, 'change:testResultTab', this.render);
+
+        this.settings = settings;
+    }
+
+
+    @on('click .tree__info')
+    onInfoClick() {
+        const show = this.settings.isShowGroupInfo();
+        this.settings.setShowGroupInfo(!show);
+    }
+
+    onRender() {
+        this.showChildView('content', new TreeView({
+            state: this.state,
+            routeState: this.routeState,
+            tabName: this.tabName,
+            baseUrl: this.baseUrl,
+            settings: this.settings,
+            collection: this.collection
+        }));
+
+        this.showChildView('search', new NodeSearchView({
+            settings: this.settings
```

Comment:
Instead of settings, the `state` modal should be used. Then you will not need to introduce an extra `saveTemporary` method.

Reply:
I agree that extending `LocalStorageModel` is not the best way, but on the other side we store all settings(and searchQuery and filtering options are the same level settings) in one place. 
But i'll do as you say.

Reply:
We are doing different separation of concerns here:
* settings is for all data, that should be saved as user preference
* state is a current state of UI (opened tree nodes, selected testcase, now is also search). It will not be saved in LocalStorage, but can be reflected in the URL, for example